packages/backend-core/src/context/mainContext.ts

```diff
@@ -123,7 +123,7 @@ export async function doInAutomationContext<T>(params: {
   task: () => T
 }): Promise<T> {
   await ensureSnippetContext()
-  return newContext(
+  return await newContext(
```

Comment:
I'm noticing in DataDog that our automation traces look incomplete, and part of me is wondering if it's because we're not awaiting some promises. This seemed like a potential culprit, and should be a semantically safe change.